### PR TITLE
Will now print summaries even when not training or when training is over

### PR DESCRIPTION
**Pull request by vincentpierre**
- [x] Requires testing with multi-brain

### CODE REVIEW

File: python/unitytrainers/trainer.py [hunk 1]
```diff
@@ -135,16 +135,18 @@ def write_summary(self, lesson_number):
         Saves training statistics to Tensorboard.
         :param lesson_number: The lesson the trainer is at.
         """
-        if (self.get_step % self.trainer_parameters['summary_freq'] == 0 and self.get_step != 0 and
-                self.is_training and self.get_step <= self.get_max_steps):
+        if self.get_step % self.trainer_parameters['summary_freq'] == 0 and self.get_step != 0:
+            training = self.is_training and self.get_step <= self.get_max_steps
```
Comment (mmattar) [inline python/unitytrainers/trainer.py]: I'd recommend a more descriptive variable name - we have `training` and `is_training`.

File: python/unitytrainers/trainer.py [hunk 2]
```diff
@@ -135,16 +135,18 @@ def write_summary(self, lesson_number):
...
             if len(self.stats['cumulative_reward']) > 0:
                 mean_reward = np.mean(self.stats['cumulative_reward'])
-                logger.info("{}: {}: Step: {}. Mean Reward: {:0.3f}. Std of Reward: {:0.3f}."
+                logger.info(" {}: {}: Step: {}. Mean Reward: {:0.3f}. Std of Reward: {:0.3f}. {}"
                             .format(self.run_id, self.brain_name, self.get_step,
-                                    mean_reward, np.std(self.stats['cumulative_reward'])))
+                                    mean_reward, np.std(self.stats['cumulative_reward']),
+                                    "Training." if training else "Not Training."))
```
Comment (mmattar) [inline python/unitytrainers/trainer.py]: nesting conditional within a logging statement makes it less readable. I would create a separate variable before the `if` command and use it here and on line 149.

File: python/unitytrainers/trainer.py [hunk 3]
```diff
@@ -135,16 +135,18 @@ def write_summary(self, lesson_number):
...
             else:
-                logger.info("{}: {}: Step: {}. No episode was completed since last summary."
-                            .format(self.run_id, self.brain_name, self.get_step))
+                logger.info(" {}: {}: Step: {}. No episode was completed since last summary. {}"
+                            .format(self.run_id, self.brain_name, self.get_step,
+                                    "Training." if training else "Not Training."))
```
Comment (mmattar) [inline python/unitytrainers/trainer.py]: Same as above.

**Comment by awjuliani** [comment]
@vincentpierre Summary printing still does not work when running soccer environment from the editor.

**Comment by vincentpierre** [comment]
@awjuliani will investigate

**Comment by vincentpierre** [comment]
@awjuliani It works for me. Both with and without `--train`.  But it takes 30 seconds for the first summary to appear. Did you pull ?

**Pull request merged**